app/src/main/java/com/github/se/wanderpals/model/viewmodel/SuggestionsViewModel.kt

```diff
@@ -188,6 +203,43 @@ open class SuggestionsViewModel(
     hideBottomSheet()
   }
 
+  /**
+   * Checks if a suggestion of a trip has reached the majority of likes and adds it as a stop if so.
+   *
+   * @param suggestion The suggestion of a trip to check and add as a stop.
+   */
+  open fun checkAndAddSuggestionAsStop(suggestion: Suggestion) {
+    viewModelScope.launch {
+      val allUsers = suggestionRepository?.getAllUsersFromTrip(tripId) ?: emptyList()

```

Comment:
You can remove the database fetch from the checkAndAddSuggestionAsStop so that it isn't called several times in the foreach loop of the loadSuggestion function. So you can put it outside of the function to have only one fetch

Reply:
Done, thank you!